lib/matplotlib/cbook.py

```diff
@@ -492,17 +504,23 @@ def connect(self, s, func):
         func will be called
         """
         self._func_cid_map.setdefault(s, WeakKeyDictionary())
-        if func in self._func_cid_map[s]:
```

Comment:
P.S. as far as I can make out, everything to do with ``weakref.ref`` you can find here.

https://hg.python.org/cpython/rev/1fb6672b20ef